entity-framework/core/modeling/relationships/one-to-one.md

```diff
@@ -0,0 +1,683 @@
+        }
+
+        // Dependent (child)
+        public class BlogHeader
+        {
+            public int Id { get; set; }
+            public int BlogId { get; set; }                                    // Required foreign key property
+            public Blog Blog { get; set; } = null!;                            // Required reference navigation to principal
+        }
+-->
+[!code-csharp[RequiredWithoutCascadeDelete](../../../../samples/core/Modeling/Relationships/OneToOne.cs?name=RequiredWithoutCascadeDelete)]
+
+By convention, required relationships are configured to [cascade delete](xref:core/saving/cascade-delete). This is because the dependent cannot exist in the database once the principal has been deleted. The database can be configured to generate an error, typically crashing the application, instead of automatically deleting dependent rows that can no longer exist. This requires some configuration:
+
+<!--
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                modelBuilder.Entity<Blog>()
+                    .HasOne(e => e.Header)
+                    .WithOne(e => e.Blog)
+                    .OnDelete(DeleteBehavior.Restrict);
+            }
+-->
+[!code-csharp[RequiredWithoutCascadeDeleteConfig](../../../../samples/core/Modeling/Relationships/OneToOne.cs?name=RequiredWithoutCascadeDeleteConfig)]
+
+## Self-referencing one-to-one
+
+In all the previous examples, the principal entity type was different from the dependent entity type. This does not have to be the case. For example, in the types below, each `Person` is optionally related to another `Person`.
+
+<!--
+        public class Person
+        {
+            public int Id { get; set; }
+
+            public int? HusbandId { get; set; }      // Optional foreign key property
+            public Person? Husband { get; set; }     // Optional reference navigation to principal
+            public Person? Wife { get; set; }        // Reference navigation to dependent
+        }
+-->
+[!code-csharp[SelfReferencing](../../../../samples/core/Modeling/Relationships/OneToOne.cs?name=SelfReferencing)]
+
+This relationship is [discovered by convention](xref:core/modeling/relationships/conventions). For cases where the navigations, foreign key, or required/optional nature of the relationship are not discovered by convention, these things can be configured explicitly. For example:
+
+<!--
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                modelBuilder.Entity<Person>()
+                    .HasOne(e => e.Husband)
+                    .WithOne(e => e.Wife)
+                    .HasForeignKey<Person>(e => e.HusbandId)
+                    .IsRequired(false);
+            }
+-->
+[!code-csharp[SelfReferencingConfig](../../../../samples/core/Modeling/Relationships/OneToOne.cs?name=SelfReferencingConfig)]
```

Comment:
Add a note saying that for self-ref the order of navigations is important. The first navigation will be considered the one pointing to the principal, therefore the FK properties will be associated with the first navigation.